src/WebApi/DependencyInjection.cs

```diff
@@ -1,18 +1,50 @@
-﻿using SSW.CleanArchitecture.Application.Common.Interfaces;
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using SSW.CleanArchitecture.Application.Common.Interfaces;
 using SSW.CleanArchitecture.Infrastructure.Persistence;
+using SSW.CleanArchitecture.WebApi.HealthChecks.EntityFrameworkDbContextHealthCheck;
 using SSW.CleanArchitecture.WebApi.Services;
 
 namespace SSW.CleanArchitecture.WebApi;
 
 public static class DependencyInjection
 {
-    public static void AddWebApi(this IServiceCollection services)
+    public static void AddWebApi(this IServiceCollection services, IConfiguration config)
     {
         services.AddHttpContextAccessor();
         services.AddScoped<ICurrentUserService, CurrentUserService>();
 
         services.AddHealthChecks()
-            .AddDbContextCheck<ApplicationDbContext>();
+            // Check 1: Check the SQL Server Connectivity (no EF, no DBContext, hardly anything to go wrong)
```

Comment:
Is it possible to move the health check code into the Infrastructure Layer?  That seems like the most appropriate place for it to live.

Reply:
May not be possible if everything depends on asp.net core though.

Reply:
yes my sentiment exactly. When i moved it into infra i brought a whole shoppingcart of dependencies along :) so i reverted.